app/cruds/cruds_elocaps.py

```diff
@@ -0,0 +1,253 @@
+            .selectinload(models_elocaps.Game.game_players)
+            .selectinload(models_elocaps.GamePlayer.player)
+            .selectinload(models_elocaps.Player.user)
+        )
+    )
+    return [i.game for i in result.scalars().all()]
+
+
+async def get_game_details(
+    db: AsyncSession, game_id: str
+) -> models_elocaps.Game | None:
+    result = await db.execute(
+        select(models_elocaps.Game)
+        .where(models_elocaps.Game.id == game_id)
+        .options(
+            selectinload(models_elocaps.Game.game_players)
+            .selectinload(models_elocaps.GamePlayer.player)
+            .selectinload(models_elocaps.Player.user)
+        )
+    )
+    return result.scalars().first()
+
+
+async def register_game(
+    db: AsyncSession,
+    game: models_elocaps.Game,
+    game_players_info: list[schemas_elocaps.GamePlayerBase],
+) -> None:
+    try:
+        db.add(game)
+        await db.commit()
+        for i in game_players_info:
+            player = (
+                (
+                    await db.execute(
+                        select(models_elocaps.Player).where(
+                            (models_elocaps.Player.user_id == i.user_id)
+                            & (models_elocaps.Player.mode == game.mode)
+                        )
+                    )
+                )
+                .scalars()
+                .first()
+            )
+            if not player:
+                player = models_elocaps.Player(user_id=i.user_id, mode=game.mode)
+                db.add(player)
+                await db.commit()
+            game_player = models_elocaps.GamePlayer(
+                game_id=game.id, player_id=player.id, team=i.team, quarters=i.quarters
+            )
+            db.add(game_player)
+            await db.commit()
+        complete_game = (
+            (
+                await db.execute(
+                    select(models_elocaps.Game)
+                    .where(models_elocaps.Game.id == game.id)
+                    .options(
+                        selectinload(models_elocaps.Game.game_players).selectinload(
+                            models_elocaps.GamePlayer.player
+                        )
+                    )
+                )
+            )
+            .scalars()
+            .first()
+        )
+        if complete_game is None:
+            raise Exception("The game that was there just disappeared, we are screwed")
+        else:
+            game = complete_game
+        for game_player in game.game_players:
+            team = game_player.team
+            c = game.get_team_quarters(~team + 4)
+            s = game.get_team_quarters(team)
+            a = game_player.quarters
+            f = a * (1 - c / s) / (c + s)
+            k = 15 * (2 - exp(-(f**2)))
+            w = game.get_winner_team() == team
+            d = game.get_team_elo(team) - game.get_team_elo(~team + 4)
+            game_player.elo_gain = int(k * (w - 1 / (1 + 10 ** (-d / 400))))
```

Comment:
I think it could be better to have a function to do these calculations. The goal of this file it to handle crud, not to generate the value to put in the database